test/jdk/java/lang/StableValue/StableValuesSafePublicationTest.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+ * @summary Basic tests for making sure StableValue publishes values safely
+ * @modules java.base/jdk.internal.misc
+ * @compile --enable-preview -source ${jdk.version} StableValuesSafePublicationTest.java
+ * @run junit/othervm --enable-preview StableValuesSafePublicationTest
+ */
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+final class StableValuesSafePublicationTest {
+
+    private static final int SIZE = 100_000;
+    private static final int THREADS = Runtime.getRuntime().availableProcessors();
+    private static final StableValue<Holder>[] STABLES = stables();
+
+    static StableValue<Holder>[] stables() {
+        @SuppressWarnings("unchecked")
+        StableValue<Holder>[] stables = (StableValue<Holder>[]) new StableValue[SIZE];
+        for (int i = 0; i < SIZE; i++) {
+            stables[i] = StableValue.of();
+        }
+        return stables;
+    }
+
+    static final class Holder {
+        // These are non-final fields but should be seen
+        // fully initialized thanks to the HB properties of StableValue.
+        int a;
+        int b;
```

Comment:
Might be worth putting in some more fields to increase the chance of detecting a mismatch.

Reply:
Good idea!